doc/cookbook/type/create_custom_field_type.rst

```diff
@@ -0,0 +1,215 @@
+create a custom field type for a specific purpose. This recipe assumes
+you need a field definition which holds specially formatted Client ID's,
+based on the existing integer field. This section explains how the field
+is defined, and how you can register it for usage in your application.
+
+The Client ID begins with a 'C' prefix, and is prepended with zeros until it's
+at least 4 digits.
+
+Example: ``C30320, C0001, C442482, C0020``.
+
+Defining the Field Type
+-----------------------
+
+In order to create the custom field type, first you have to create the class representing the type.
+In this situation the class holding the field type will be called ClientIdType and the file will be
+stored in the default location for search field types, which is ``<VendorName>\\Search\\Type``.
+Make sure the field extends :class:`Rollerworks\\Component\\Search\\AbstractFieldType`:
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/Client/Search/Type/ClientIdType.php
+
+    namespace Acme\Client\Search\Type;
+
+    use Acme\Client\Search\DataTransformer\ClientIdTransformer;
+    use Rollerworks\Component\Search\AbstractFieldType;
+    use Symfony\Component\OptionsResolver\OptionsResolver;
+
+    class ClientIdType extends AbstractFieldType
+    {
+        public function buildType(FieldConfigInterface $config, array $options)
+        {
+            // The integer-type sets a special transformer for localized integers
+            // This type doesn't need this so remove the transformers
+            $config->resetViewTransformers();
+
+            $config->addViewTransformer(new ClientIdTransformer());
+        }
+
+        public function configureOptions(OptionsResolver $resolver)
+        {
+            $resolver->setDefaults(
+                array(
+                    'precision' => 0
+                )
+            );
+        }
+
+        public function getName()
+        {
+            return 'client_id';
+        }
+
+        public function getParent()
+        {
+            return 'integer';
+        }
+    }
+
+.. tip::
+
+    The location of this file is not important - the ``Search\\Type`` directory
+    is just a convention.
+
+Here, the return value of the ``getParent`` function indicates that you're
+extending the ``integer`` field type. This means that, by default, you inherit
+all of the logic and rendering of that field type. To see some of the logic,
+check out the `IntegerType`_ class. There are three methods that are particularly
+important:
+
+``buildType()``
+    Each field type has a ``buildType`` method, which is where
+    you configure and build any field(s).
+
+``buildView()``
+    This method is used to set any extra variables you'll
+    need when rendering your field in a template. For example, in `IntegerType`_,
+    a ``precision`` variable is set and used in the template to set
+    the ``precision`` attribute on the ``input`` field.
+
+``configureOptions()``
+    This defines options for your field type that
```

Comment:
Copy paste from the Symfony docs, I cut lines at 72 characters for sentences but words continue on the same line.